File: Kotlin/그리디 알고리즘/2847.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val N = readLine().toInt()
    val arr = ArrayList<Int>()
    var res = 0

    for (i in 0 until N){
        arr.add(readLine().toInt())
    }

    for (i in N-2 downTo 0){
        if (arr[i] >= arr[i+1]){
            val temp = arr[i] - arr[i+1] + 1
            res += temp
            arr[i] -= temp
        }
    }

    println(res)
}

File: BOJ/Python/그리디 알고리즘/2847.py
N = int(input())
arr = list()
res = 0

for _ in range(N):
    arr.append(int(input()))

for i in range(N-2, -1, -1):
    if arr[i] >= arr[i+1]:
        temp = arr[i] - arr[i+1] + 1
        res += temp
        arr[i] -= temp

print(res)